lucille-core/src/main/java/com/kmwllc/lucille/stage/AddRandomField.java

```diff
@@ -0,0 +1,155 @@
+  private final boolean isNested;
+  private List<String> dataArr;
+  private List<String> uniqueValues;
+
+  public AddRandomField(Config config) throws StageException {
+    super(config, new StageSpec().withOptionalProperties("input_data_path", "field_name", "cardinality", "min_num_of_terms",
+        "max_num_of_terms", "is_nested"));
+    this.inputDataPath = ConfigUtils.getOrDefault(config, "input_data_path", null);
+    this.fieldName = ConfigUtils.getOrDefault(config, "field_name", "data");
+    this.minNumOfTerms = ConfigUtils.getOrDefault(config, "min_num_of_terms", null);
+    this.maxNumOfTerms = ConfigUtils.getOrDefault(config, "max_num_of_terms", null);
+    this.isNested = ConfigUtils.getOrDefault(config, "is_nested", false);
+    this.dataArr = null;
+    this.cardinality = null;
+    this.uniqueValues = null;
+
+    if (this.minNumOfTerms == null ^ this.maxNumOfTerms == null) {
+      throw new StageException("Both minimum and maximum number of terms must be specified");
+    }
+    if (this.minNumOfTerms == null && this.maxNumOfTerms == null) {
+      this.minNumOfTerms = 1;
+      this.maxNumOfTerms = 1;
+    }
+    if (this.minNumOfTerms > this.maxNumOfTerms) {
+      throw new StageException("Minimum number of terms must be less than or equal to maximum");
+    }
+  }
+
+  @Override
+  public void start() throws StageException {
+    this.dataArr = this.inputDataPath != null ? getFileData(this.inputDataPath) : null;
+    this.cardinality = ConfigUtils.getOrDefault(config, "cardinality",
+        this.dataArr != null ? this.dataArr.size() : this.maxNumOfTerms);
+    this.uniqueValues = getUniqueValues(this.dataArr != null, this.dataArr);
+
+    if (this.dataArr != null && this.cardinality > this.dataArr.size()) {
+      throw new StageException("Cardinality must be less than the number of lines in given file");
+    }
+  }
+
+  @Override
+  public Iterator<Document> processDocument(Document doc) throws StageException {
+    ArrayList<String> fieldDataArr = genFieldDataArr(uniqueValues, minNumOfTerms, maxNumOfTerms);
+    if (isNested) {
+      populateFieldsNested(fieldName, doc, fieldDataArr);
+    } else {
+      populateFieldsDefault(fieldName, doc, fieldDataArr);
+    }
+    return null;
+  }
+
+  private void populateFieldsNested(String fieldName, Document doc, ArrayList<String> fieldDataArr) {
+    ObjectMapper mapper = new ObjectMapper();
+    ArrayNode array = mapper.createArrayNode();
+    for (String value : fieldDataArr) {
+      array.add(mapper.createObjectNode().put("data", value));
+    }
+    doc.setField(fieldName, array);
+  }
+
+  private void populateFieldsDefault(String fieldName, Document doc, ArrayList<String> fieldDataArr) {
+    for (String value : fieldDataArr) {
+      doc.setOrAdd(fieldName, value);
+    }
+  }
+
+  private List<String> getFileData(String inputDataPath) throws StageException {
+    List<String> data = null;
+    try {
+      data = Files.readAllLines(Path.of(inputDataPath));
```

Comment:
you can put the return here, inside the try
that elimates the declaration above and the return below

Reply:
done